packages/scaffold-config/src/frameworks.ts

```diff
@@ -1,689 +1,253 @@
+  try {
+    const loc = resolveFrom(projectPath, path.join(dependency.package, 'package.json'))
+    const pkg = fs.readJsonSync(loc) as PkgJson
+    const pkgVersion = semver.coerce(pkg.version)
+
+    if (!pkgVersion) {
+      throw Error(`${pkg.version} for ${dependency.package} is not a valid semantic version.`)
+    }
+
+    return {
+      dependency,
+      detectedVersion: pkg.version,
+      loc,
+      satisfied: Boolean(pkg.version && semver.satisfies(pkgVersion, dependency.minVersion)),
+    }
+  } catch (e) {
+    return {
+      dependency,
+      detectedVersion: null,
+      loc: null,
+      satisfied: false,
+    }
+  }
+}
+
+function getBundlerDependency (bundler: WizardBundler, projectPath: string): DependencyToInstall {
+  switch (bundler) {
+    case 'vite': return inPkgJson(dependencies.WIZARD_DEPENDENCY_VITE, projectPath)
+    case 'webpack': return inPkgJson(dependencies.WIZARD_DEPENDENCY_WEBPACK, projectPath)
+    default: throw Error(`Unknown bundler ${bundler}`)
+  }
+}
+
+interface CreateCypressConfig {
+  framework: typeof WIZARD_FRAMEWORKS[number]['configFramework']
+  bundler: WizardBundler
+  language: 'js' | 'ts'
+  projectRoot: string
+}
+
+export function createCypressConfig (config: CreateCypressConfig): string {
+  if (config.language === 'ts') {
+    if (defineConfigAvailable(config.projectRoot)) {
+      return dedent`
+        import { defineConfig } from 'cypress'
 
-const isWebpack = (bundler: Bundler) => ['webpack4', 'webpack5'].includes(bundler)
-
-export const FRONTEND_FRAMEWORKS = [
-  {
-    type: 'crav4',
-    family: 'template',
-    name: 'Create React App (v4)',
-    supportedBundlers: [BUNDLER_WEBPACK_4],
-    packages: [
-      CYPRESS_REACT_LATEST,
-      BUNDLER_WEBPACK_4,
-    ],
-    defaultPackagePath: '@cypress/react/plugins/react-scripts',
-    glob: '*.{js,jsx,tsx}',
-    category: FRONTEND_FRAMEWORK_CATEGORIES[0],
-    codeGenFramework: CODE_GEN_FRAMEWORKS[0],
-    storybookDep: STORYBOOK_REACT,
-    detectors: [
-      {
-        dependency: 'react-scripts',
-        version: '^4.0.0',
-      },
-    ],
-    config: {
-      js: () => {
-        return dedent`
-      const { devServer } = require('@cypress/react/plugins/react-scripts')
+        export default defineConfig({
+          component: {
+            devServer: {
+              framework: '${config.framework}',
+              bundler: '${config.bundler}'
+            }
+          }
+        })`
+    }
 
-      module.exports = {
+    return dedent`
+      export default {
         component: {
-          devServer,
-        },
+          devServer: {
+            framework: '${config.framework}',
+            bundler: '${config.bundler}'
+          }
+        }
       }`
-      },
-      ts: () => {
-        return dedent`
-      import { defineConfig } from 'cypress'
-      import { devServer } from '@cypress/react/plugins/react-scripts'
+  }
+
+  return dedent`
```

Comment:
The `cypress.config.js` should also use `defineConfig` if it is available.

Reply:
Did not realise. I will make this change at the end once all the presets are merged, along with updating the dependencies based on whatever decision we arrive at.